src/main/java/io/github/bobocodebreskul/context/support/BeanDefinitionReaderUtils.java

```diff
@@ -35,32 +38,43 @@ public static String generateBeanName(BeanDefinition beanDefinition,
       log.error("Failed to generate bean name for nullable bean definition");
       return "Null bean definition specified";
     });
-    Objects.requireNonNull(beanDefinition.getBeanClass(), () -> {
-      log.error("Failed to generate bean name for nullable bean class");
-      return "Bean class has not been specified";
-    });
+    validateBeanClassNonNull(beanDefinition.getBeanClass());
+    validateBeanClassName(beanDefinition, registry);
 
-    Class<?> beanClass = beanDefinition.getBeanClass();
-    String beanName = StringUtils.uncapitalize(beanClass.getSimpleName());
-    if (isSameBeanNameFromAnotherPackage(registry, beanDefinition, beanName)) {
-      beanName = beanClass.getName();
-    }
+    String beanName = uncapitalize(beanDefinition.getBeanClass());
+
+    log.trace("Generated bean name: {} for class {}", beanName, beanDefinition.getBeanClass().getName());
 
-    log.trace("Generated bean name: {} for class {}", beanName, beanClass.getName());
     return beanName;
   }
 
   /**
-   * Returns a list of bean types that this bean depends on, including constructor argument types,
-   * fields, and methods annotated with the {@link Autowired} annotation.
+   * Generate bean name for the given class type.
+   *
+   * @param beanClass bean class to generate bean name for
+   * @return the generated bean name
+   */
+  public static String generateClassBeanName(Class<?> beanClass) {
+    validateBeanClassNonNull(beanClass);
+    return uncapitalize(beanClass);
+  }
+
+  /**
+   * Returns a list of bean dependencies (as {@link BeanDependency}) this bean depends on,
+   * including constructor argument types, fields, and methods annotated with the
+   * {@link Autowired} annotation.
    *
    * @param beanClass the class of the bean to scan for dependencies
    * @return a list of bean dependency types
    */
-  public static List<Class<?>> getBeanDependencies(Class<?> beanClass) {
+  public static List<BeanDependency> getBeanDependencies(Class<?> beanClass) {
     log.trace("Scanning class {} for @Autowire candidates", beanClass.getName());
-    Stream<Class<?>> constructorDependenciesStream = Arrays.stream(
-            beanClass.getDeclaredConstructors())
+
+    Constructor<?>[] declaredConstructors = beanClass.getDeclaredConstructors();
+
+    validateBeanClassConstructorSize(beanClass.getName(), declaredConstructors);
+
+    Stream<Class<?>> constructorDependenciesStream = Arrays.stream(declaredConstructors)
```

Comment:
Do we need stream there? if we expect exactly one constructor currently maybe it redundant there and add over logic and miss understanding for reader

Reply:
This is old changes that would be changed in the next pull request after this one will have been merged. It will require some additional refactoring changes, so I propose  not to do them here.